delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/multifetch/MultiFetch.scala

```diff
@@ -0,0 +1,53 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.multifetch
+
+import ch.epfl.bluebrain.nexus.delta.sdk.acls.AclCheck
+import ch.epfl.bluebrain.nexus.delta.sdk.identities.model.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdContent
+import ch.epfl.bluebrain.nexus.delta.sdk.multifetch.model.MultiFetchResponse.Result
+import ch.epfl.bluebrain.nexus.delta.sdk.multifetch.model.MultiFetchResponse.Result._
+import ch.epfl.bluebrain.nexus.delta.sdk.multifetch.model.{MultiFetchRequest, MultiFetchResponse}
+import ch.epfl.bluebrain.nexus.delta.sdk.permissions.Permissions.resources
+import monix.bio.UIO
+
+/**
+  * Allows to fetch multiple resources of different types in one request.
+  *
+  * The response includes a resources array that contains the resources in the order specified in the request. If there
+  * is a failure getting a particular resource, the error is included in place of the resource.
+  */
+trait MultiFetch {
+
+  def apply(request: MultiFetchRequest)(implicit caller: Caller): UIO[MultiFetchResponse]
+
+}
+
+object MultiFetch {
+  def apply(
+      aclCheck: AclCheck,
+      fetchResource: MultiFetchRequest.Input => UIO[Option[JsonLdContent[_, _]]]
+  ): MultiFetch =
+    new MultiFetch {
+      override def apply(request: MultiFetchRequest)(implicit
+          caller: Caller
+      ): UIO[MultiFetchResponse] = {
+        val fetchAllCached = aclCheck.fetchAll.memoizeOnSuccess
+        request.resources
+          .traverse { input =>
+            for {
+              authorized <- aclCheck.authorizeFor(input.project, resources.read, fetchAllCached)
+              resource   <- if (authorized) fetchResource(input) else UIO.none
+            } yield {
+              if (authorized) resource.fold[Result](NotFound(input.id, input.project)) { content =>
+                Success(input.id, input.project, content)
+              }
```

Comment:
I know this is personal preference, but `.getOrElse` is much clearer intent than `fold`ing on an Option